### DISCUSSION TITLE
Bind() to 3rd ObservableCollection

### DISCUSSION THREAD

**Discussion by a44281071** [reactivemarbles/DynamicData#602] [Q&A] [answered]
I wan't bind to Caliburn.Micro.BindableCollection<T>

``` csharp
        private IDisposable CreateObserverHandle()
        {
            var material_handle = all_materials.Connect()
                                  .OnItemAdded(dd => Trace.TraceInformation("Materials add item: {0}", dd.Id))
                                  .Bind(Items as BindableCollection<MaterialBoxViewModel>)
                                  .Subscribe();
            return material_handle;
        }
```

**Comment by glennawatson** [comment, marked as answer]
There are no plans to support binding to a BindableCollection

**Reply by RolandPheasant** [reply]
You should be able to write an adaptor to do this.

See  https://github.com/reactivemarbles/DynamicData/blob/9e17d5e7d2e979b84fd602fcfe2f48d3c22609e3/src/DynamicData/Binding/ObservableCollectionAdaptor.cs#L16 for how DD does it to populate an observable collection from an observable change set.

Clone the DD code above, and adapt and change the parameter `IObservableCollection<T>`  to  `BindableCollection<T>`.

Then you can use something like:

```
    var material_handle = all_materials.Connect()
                                  .OnItemAdded(dd => Trace.TraceInformation("Materials add item: {0}", dd.Id))
                                  .Adapt(new MyBindableCollectionAdaptor<MaterialBoxViewModel>(Items))
                                  .Subscribe();
```

The example is for a List but the Cache does it a little differently. Grab the source code and check it out for yourself.
